cloudsdk/src/main/java/io/particle/android/sdk/cloud/ParticleDevice.java

```diff
@@ -406,6 +406,31 @@ public void executeFlashingChange() throws RetrofitError {
         });
     }
 
+    @WorkerThread
+    public void flashCodeFile(final File file) throws ParticleCloudException {
+        performFlashingChange(new FlashingChange() {
+            @Override
+            public void executeFlashingChange() throws RetrofitError {
+                mainApi.flashFile(deviceState.deviceId,
+                        new TypedFile("multipart/form-data", file));
+            }
+        });
+    }
+
+
+    @WorkerThread
```

Comment:
Ha!  I'd forgotten we had both of those methods for the binary flash methods.

Anyway, between the use case you cited and the desire for API symmetry between these methods and the flashBinary...() methods, I think this should be included after all.